lib/page_setup.rb

```diff
@@ -0,0 +1,94 @@
+require 'tools'
+
+class PageSetup
```

Comment:
again, i see where you're going with this - but i think you may have over-complicated this. it's okay to repeat some code to simplify this 👍